README.md

```diff
@@ -71,13 +81,48 @@ This RFC makes no requirement as to how the transition should happen, but here a
 # Examples
 [examples]: #examples
 
-- `pkgs.hello`:
-  - Move from `pkgs/applications/misc/hello/default.nix` to `pkgs/unit/hello/pkg-fun.nix`
-  - Move from `pkgs/applications/misc/hello/test.nix` to `pkgs/unit/hello/test.nix`
-- `pkgs.gnumake`: Move from `pkgs/development/tools/build-managers/gnumake` to `pkgs/unit/gnumake`
-- `pkgs.gnumake42`: Move from `pkgs/development/tools/build-managers/gnumake/4.2` to `pkgs/unit/gnumake42`
-- `pkgs.buildEnv`: Move from `pkgs/build-support/buildenv` to `pkgs/unit/buildEnv`
-- `pkgs.fetchFromGitHub`: Move from `pkgs/build-support/fetchgithub` to `pkgs/unit/fetchFromGitHub`
+- `pkgs.hello` matches all criteria:
+  The default Nix file [`pkgs/applications/misc/hello/default.nix`](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/applications/misc/hello/default.nix) only transitively [references `test.nix`](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/applications/misc/hello/default.nix#L31) in the same directory.
+  Neither the `default.nix` nor `test.nix` is referenced by any other file in nixpkgs, so we can do the transformation:
+  - Move `pkgs/applications/misc/hello/default.nix` to `pkgs/unit/hell/hello/pkg-fun.nix`
+  - Move `pkgs/applications/misc/hello/test.nix` to `pkgs/unit/hell/hello/???` [TODO](https://github.com/nixpkgs-architecture/simple-package-paths/issues/19)
+- `pkgs.gnumake` matches all criteria:
+  The default Nix file [`pkgs/development/tools/build-managers/gnumake/default.nix`](https://github.com/NixOS/nixpkgs/blob/nixos-22.05/pkgs/development/tools/build-managers/gnumake/default.nix) transitively references only files in its own directory and no other files in nixpkgs reference `gnumake`'s files, so we can do the transformation by moving all the files from `pkgs/development/tools/build-managers/gnumake` to `pkgs/unit/gnum/gnumake`, the default Nix file ending up in `pkgs/unit/gnum/gnumake/pkg-fun.nix`.
+- Similarly `pkgs.gnumake42` in [`pkgs/development/tools/build-managers/gnumake/4.2/default.nix`](https://github.com/NixOS/nixpkgs/tree/nixos-22.05/pkgs/development/tools/build-managers/gnumake/4.2/default.nix) fulfils all criteria, even though its directory is nested in `pkgs.gnumake`'s directory, they don't reference each others files.
+- `pkgs.zsh` matches all criteria, its default Nix file is moved to `pkgs/unit/zsh-/zsh/pkg-fun.nix`
+- `pkgs.emptyFile` doesn't fulfil criteria 1 (it's defined in `pkgs/build-support/trivial-builders.nix`) and 2 (it's defined inline), so it can't be moved into `pkgs/unit`
+- `pkgs.fetchFromGitHub` doesn't fulfil the criteria 3 (it evaluates to a function), so it can't be moved into `pkgs/unit`
+
+Here's how such a `pkgs/unit` directory structure would look like, note how all attribute names have the same level of nesting:
+```
+pkgs
+└── unit
+   ├── acpi
+   │  ├── acpi
+   │  ├── acpica-tools
+   │  ├── acpid
+   │  ┊
+   ┊
+   ├── auto
+   │  ├── autossh
+   │  ├── automirror
+   │  ├── autosuspend
+   │  ┊
+   ┊
+   ├── sl--
```

Comment:
Why not `substring 0 4 "sl"`?

Likewise for zsh.

Reply:
Allow me to paint a picture. 

Howard has been using Nix for one year. He excitedly makes his first PR to Nixpkgs.

The path names seem weird :

"foo/foo/default.nix"
"u/u/default.nix"
"emac/emacs/default.nix"

Howard is confused about the max length but tries his best, he PRs "libxz/libxz/default.nix". Howard gets an email requesting that he refilehis PR with a new name, he rolls his eyes "how was I supposed to guess that?", but tells himself he'll refile when he has some free time.

It is 6 months later. The seasons have changed, but Howard's PR has not. Stale Bot adds yet another notification to Robert's feed of stale PRs - left dying on the vine.

Nixpkgs is in shambles! If only the maintainers had chosen `jq--/jq/default.nix` this all could have been avoided!

</scene>

Reply:
[Another comment by @aakropotkin on Discourse](https://discourse.nixos.org/t/2022-11-21-nixpkgs-architecture-team-meeting-18/23393/4?u=infinisil)

Originally in the meeting we used `--` in case there's multiple levels of sharding, like

```
he/ll/hello
zs/h-/zsh
sl/--/sl
r-/--/r
```

This makes it clear that e.g. `sl` shouldn't be in `sl/sl/pkg-fun.nix` where it would collide with a package named `slsl` in `sl/sl/slsl/pkg-fun.nix`. `sl` being in `sl/--/sl/pkg-fun.nix` gets rid of the ambiguity.

However, later in the meeting we figured out that there's no good reason to have nested folders, so we changed it to 
```
```
hell/hello
zsh-/zsh
sl--/sl
r---/r
```

instead, which lines up very nicely still. But you're right roberth, with this scheme there's no super good reason for the `-` anymore.

I'm 50/50 on this, because as @aakropotkin mentions, the `-` do help with clarity, establishing that it's a 4-letter prefix, and not just the package name. Of course we could enforce either option with CI checks (and we totally should).

Reply:
> The path names seem weird :
> 
> "foo/foo/default.nix"
> "u/u/default.nix"
> "emac/emacs/default.nix"

This is not realistic.
Only 1 in 22 packages is shorter than 4 characters, and only 1 in 8 isn't truncated.
Let's avoid solving problems that don't exist.

> CI checks (and we totally should).

I agree. We don't want to have any drift from the sharding specification, because that would defeat the point.

Reply:
IMO filling up shorter packages with dashes is ugly. Can't we just trim them to a minimal length and keep them as is if they are not longer?

Reply:
@SuperSandro2000 Yes that's @roberth's proposal we're discussing here

Reply:
I updated the PR to use this proposal now